src/mlpack/methods/preprocess/preprocess_split_main.cpp

```diff
@@ -148,22 +164,45 @@ static void mlpackMain()
         IO::GetParam<arma::Mat<size_t>>("input_labels");
     arma::Row<size_t> labelsRow = labels.row(0);
 
-    const auto value = data::Split(data, labelsRow, testRatio, !shuffleData);
-    Log::Info << "Training data contains " << get<0>(value).n_cols << " points."
-        << endl;
-    Log::Info << "Test data contains " << get<1>(value).n_cols << " points."
-        << endl;
-
-    if (IO::HasParam("training"))
-      IO::GetParam<arma::mat>("training") = std::move(get<0>(value));
-    if (IO::HasParam("test"))
-      IO::GetParam<arma::mat>("test") = std::move(get<1>(value));
-    if (IO::HasParam("training_labels"))
-      IO::GetParam<arma::Mat<size_t>>("training_labels") =
-          std::move(get<2>(value));
-    if (IO::HasParam("test_labels"))
-      IO::GetParam<arma::Mat<size_t>>("test_labels") =
-          std::move(get<3>(value));
+    if(stratifyData)
+    {
+      const auto value =
+          data::StratifiedSplit(data, labelsRow, testRatio, !shuffleData);
+          Log::Info << "Training data contains "
+              << get<0>(value).n_cols << " points." << endl;
+          Log::Info << "Test data contains "
+              << get<1>(value).n_cols << " points." << endl;
+
+          if (IO::HasParam("training"))
+            IO::GetParam<arma::mat>("training") = std::move(get<0>(value));
+          if (IO::HasParam("test"))
+            IO::GetParam<arma::mat>("test") = std::move(get<1>(value));
+          if (IO::HasParam("training_labels"))
+            IO::GetParam<arma::Mat<size_t>>("training_labels") =
+                std::move(get<2>(value));
+          if (IO::HasParam("test_labels"))
+            IO::GetParam<arma::Mat<size_t>>("test_labels") =
+                std::move(get<3>(value));
```

Comment:
```suggestion
        Log::Info << "Training data contains "
            << get<0>(value).n_cols << " points." << endl;
        Log::Info << "Test data contains "
            << get<1>(value).n_cols << " points." << endl;

        if (IO::HasParam("training"))
          IO::GetParam<arma::mat>("training") = std::move(get<0>(value));
        if (IO::HasParam("test"))
          IO::GetParam<arma::mat>("test") = std::move(get<1>(value));
        if (IO::HasParam("training_labels"))
          IO::GetParam<arma::Mat<size_t>>("training_labels") =
              std::move(get<2>(value));
        if (IO::HasParam("test_labels"))
          IO::GetParam<arma::Mat<size_t>>("test_labels") =
              std::move(get<3>(value));
```
I guess the indentation is incorrect.